### ISSUE TITLE
Refactor Graying out timeslots logic

### ISSUE THREAD

**Issue by yuvrajjain2003** [unc-csxl/csxl.unc.edu#340]
## Why?

If Sally student has a drop-in reservation, then she should not be able to reserve a room during that time slot. Therefore, the table should gray out these slots to indicate this. Also, if the current time is 11:03 am, then Sally should still be able to reserve a room from 11:03 to 11:30 am. 

## Acceptance Criteria

- [ ] Round down to the nearest half hour instead of rounding up to the nearest half hour when graying out time slots based on current time.
- [ ] Time slots should be grayed out during the time when a person has a drop-in reservation to the XL.
- [ ] Figure out what is wrong with operating hours and why room reservations are opening at 10:30 am rather than 10:00 am.

**Comment by KrisJordan** [comment]
For what it is worth, hours are being added at 10:00:00 start in production and still seeing the truncation of missing the first 30 minutes. So must be a small, pesky boundary condition.

**Comment by KrisJordan** [comment]
I'm not sure if this is the right issue, but I believe we should special case the table to not show past slots, rather than gray them out. Later in the day, especially on mobile, you have to scroll to see what availability looks like right now.

**Comment by yuvrajjain2003** [comment]
> I'm not sure if this is the right issue, but I believe we should special case the table to not show past slots, rather than gray them out. Later in the day, especially on mobile, you have to scroll to see what availability looks like right now.

Yes, that was actually the goal. Graying out past timeslots was a duct-tape fix we used when the table was static, but I'll add that as one of the tasks under this PR as well!
